compiler-core/src/type_/pattern.rs

```diff
@@ -425,9 +432,27 @@ impl<'a, 'b> PatternTyper<'a, 'b> {
                 // Register the value as seen for detection of unused values
                 self.environment.increment_usage(&name);
 
+                let map_error_variables = &self.environment.local_value_names();
                 let cons = self
                     .environment
                     .get_value_constructor(module.as_ref(), &name)
+                    .map_err(|e| {
+                        if e == (UnknownValueConstructorError::Variable {
+                            name: name.clone(),
+                            variables: map_error_variables.clone(),
+                            situation: None,
+                        }) {
+                            UnknownValueConstructorError::Variable {
+                                name: name.clone(),
+                                variables: map_error_variables.clone(),
+                                situation: Some(
+                                    UnknownValueConstructorErrorSituation::NameStartsWithCapitalCase,
+                                ),
+                            }
+                        } else {
+                            e
+                        }
+                    })
```

Comment:
Extract this into a helper function like we do with unify errors please, and use pattern matching rather than equality. The equality approach here does a lot of cloning and so is slower.

Reply:
I shifted this into the ```convert_get_value_constructor_error``` function. I'd think it's fine and safe to do so.